web/app/routes/authenticated/dashboard.ts

```diff
@@ -54,13 +54,26 @@ export default class DashboardRoute extends Route {
         return result.hits as HermesDocument[];
       });
 
-    try {
+    /**
+     * If the user is loading the dashboard for the first time,
+     * we await the `fetchAll` task so we can display the
+     * documents at the same time as the rest of the layout.
+     *
+     * If a user leaves and returns to the dashboard, we call `fetchAll`
+     * but don't await it. This speeds up the transition and is especially
+     * efficient when the RecentDocs list hasn't changed since its last load.
+     *
+     * It's possible for the user to see the RecentDocs list update
+     * in real time (by visiting a document and very quickly clicking back),
+     * but unlikely, since we call `fetchAll` in the `/document` route,
+     * just after the doc is marked viewed. In most cases, the task will be
+     * finished by the time the user returns to the dashboard.
+     *
+     */
+    if (this.recentDocs.all) {
+      void this.recentDocs.fetchAll.perform();
+    } else {
       await this.recentDocs.fetchAll.perform();
-    } catch {
-      /**
-       * This tells our template to show the error state.
-       */
-      this.recentDocs.all = null;
```

Comment:
Do we still want this in case of an error when fetching recent docs?

Reply:
This is now in the service: https://github.com/hashicorp-forge/hermes/pull/318/files#diff-e4377cd5d1c050300d977e533d7234362dbf38040020709bc2188ca12951bf3bR102

And should still trigger the template error: https://github.com/hashicorp-forge/hermes/blob/main/web/app/templates/authenticated/dashboard.hbs#L49

Reply:
Ah great 👍